pkg/devserver/devserver.go

```diff
@@ -857,24 +793,36 @@ func NormalizeThrottle(smv2 sv2.StateLoader, dbcqrs cqrs.Manager) redis_state.Re
 }
 
 func PartitionConstraintConfigGetter(dbcqrs cqrs.Manager) redis_state.PartitionConstraintConfigGetter {
-	return func(ctx context.Context, p redis_state.QueueShadowPartition) (*redis_state.PartitionConstraintConfig, error) {
-		if p.EnvID == nil || p.FunctionID == nil {
-			return &redis_state.PartitionConstraintConfig{
-				Concurrency: redis_state.ShadowPartitionConcurrency{
-					SystemConcurrency:   consts.DefaultConcurrencyLimit,
-					AccountConcurrency:  consts.DefaultConcurrencyLimit,
-					FunctionConcurrency: consts.DefaultConcurrencyLimit,
+	return func(ctx context.Context, p redis_state.PartitionIdentifier) redis_state.PartitionConstraintConfig {
+		if p.SystemQueueName != nil {
+			return redis_state.PartitionConstraintConfig{
+				Concurrency: redis_state.PartitionConcurrency{
+					SystemConcurrency:   consts.DefaultSystemConcurrencyLimit,
+					AccountConcurrency:  consts.DefaultSystemConcurrencyLimit,
+					FunctionConcurrency: consts.DefaultSystemConcurrencyLimit,
 				},
-			}, nil
+			}
 		}
 
-		workflow, err := dbcqrs.GetFunctionByInternalUUID(ctx, *p.EnvID, *p.FunctionID)
+		workflow, err := dbcqrs.GetFunctionByInternalUUID(ctx, p.EnvID, p.FunctionID)
 		if err != nil {
-			return nil, fmt.Errorf("could not find workflow: %w", err)
+			return redis_state.PartitionConstraintConfig{
+				Concurrency: redis_state.PartitionConcurrency{
+					SystemConcurrency:   consts.DefaultConcurrencyLimit,
+					AccountConcurrency:  consts.DefaultConcurrencyLimit,
+					FunctionConcurrency: consts.DefaultConcurrencyLimit,
+				},
+			}
 		}
 		fn, err := workflow.InngestFunction()
 		if err != nil {
-			return nil, fmt.Errorf("could not convert workflow to inngest function: %w", err)
+			return redis_state.PartitionConstraintConfig{
```

Comment:
I'd recommend to declare a global private default config, and return that instead of repeating this multiple times.

```go
var defaultPartitionConstraintConfig = redis_state.PartitionConstraintConfig{
  Concurrency: redis_state.PartitionConcurrency{
    SystemConcurrency:   consts.DefaultConcurrencyLimit,
    AccountConcurrency:  consts.DefaultConcurrencyLimit,
    FunctionConcurrency: consts.DefaultConcurrencyLimit,
  },
}
```